[REF] Planning: create new field service page in planning (#17905)
odoo/documentation: content/applications/services/planning.rst

.. note::
   If the Planning roles fields are left empty, there are no restrictions in the shift templates and
   open shifts shared with the employee. However, it's not possible to use the **Auto Plan** feature

>    for employee with no roles.

Reviewer: ```suggestion
   open shifts shared with the employee. However, it is not possible to use the **Auto Plan** feature
for employees with no roles.
```

    gradually. To change the time frame, :ref:`activate the developer mode <developer-mode>`, then
    go to :menuselection:`Planning --> Configuration --> Settings` and edit the :guilabel:`Recurring
    Shifts`.

- :guilabel:`Allocated Time`: Is calculated based on the shift's date and time, and the employee's
  :guilabel:`Working Schedule`.
- :guilabel:`Role`: Select the role that the assigned resources will be performing. This field is
  especially useful when :ref:`auto-planning <planning/open-shifts>` shifts. Once you select a role,
  the :ref:`shift templates <planning/templates>` associated with it are displayed in the upper
  section of the pop-up window.

Additionally, the following fields may also be available:

- :guilabel:`Project`: If the :guilabel:`Project planning` setting is enabled in the Planning
  settings, select the relevant project to link to the shift. Shifts dedicated to the selected
  project are then planned and tracked accordingly.
- :guilabel:`Task`: If a project is selected, optionally select a task within that project.
- :guilabel:`Sales Order`: If the Sales app is installed, select the relevant sales order to link to
  the shift.

> - :guilabel:`Customer`: If the shift is an :doc:`onsite intervention <planning/field_service>`,

Reviewer: ```suggestion
- :guilabel:`Customer`: If the shift is an :doc:`on-site intervention <planning/field_service>`,
```

.. image:: planning/save-template.png
   :alt: Shift form.

Alternatively, go to :menuselection:`Planning --> Configuration --> Shift Templates`, then

> click :guilabel:`New`. Fill in the :guilabel:`Start Hour` and :guilabel:`Span`. The shift's

Reviewer: Not sure if I am missing something, but when I enter the start time and a span, the end time is not calculated but remains unchanged.

Reviewer 2: Actually no this is outdated, it looks like the field label change was updated in the doc, but not the actualy functioning of the field (which changed along with the label in 18.0 it seems) I'm going to do a short edit so this isn't wrong, but I've added it to the Planning doc task because I'm not sure I understand how a shift's allocated time is calculated when created from a shift template that spans over several days ..

Planning shifts
===============

> On opening the Planning app, the users see their own schedule. Users with administrator access

Reviewer: ```suggestion
On opening the Planning app, users see their own schedule. Users with administrator access
```

rights can also see :guilabel:`Schedule by Resource`, :guilabel:`Role`, :guilabel:`Project`, or

> :guilabel:`Sales Order`, as well as reporting and configuration menus.

Reviewer: ```suggestion
rights can also see the :guilabel:`Schedule` :guilabel:`By Resource`, :guilabel:`By Role`, :guilabel:`By Project`, or
:guilabel:`By Sales Order`, as well as reporting and configuration menus.
```

.. note::
   The schedule is displayed in the Gantt view, which allows you to reassign, reschedule, resize,
   split, and duplicate shifts without having to open them.

.. image:: planning/schedule.png
   :alt: A schedule displaying various visual elements.

The following visual elements are used on the shifts in the schedules:

- **Full color**: shifts that are planned and :ref:`published <planning/shifts-publish>`.
- **Diagonal stripes**: shifts that are planned but have yet to be :ref:`published
  <planning/shifts-publish>`.

> - **Grayed-out background**: employees that are on time off.

Reviewer: I know you didn't touch this, but this bothers me
```suggestion
- **Grayed-out background**: employees who are on time off.
```

  displayed in the upper section of the pop-up window. Once selected, a template prefills the
  shift form accordingly.
- :guilabel:`Resources`: Resources can be both employees or materials. If this field is left empty,
  the shift is considered an :ref:`open shift <planning/open-shifts>`.
- :guilabel:`Date`: Choose the date and time of the shift. This is the only mandatory field when
  creating a shift.
- :guilabel:`Repeat`: Click the :icon:`fa-repeat` :guilabel:`(repeat)` button and configure the
  :guilabel:`Repeat every` fields as needed. The following rules apply to recurring shifts:

  - All fields (e.g., :guilabel:`Resource`, :guilabel:`Role`, :guilabel:`Project`) are copied from
    the original shift except for the date, which is adjusted according to the :guilabel:`Repeat
    Every` field.
  - Recurrences are planned but not published.
  - By default, planned shifts are created six months in advance, after which they are created
    gradually. To change the time frame, :ref:`activate the developer mode <developer-mode>`, then
    go to :menuselection:`Planning --> Configuration --> Settings` and edit the :guilabel:`Recurring
    Shifts`.

> - :guilabel:`Allocated time`: Is calculated based on the date and the employee's :guilabel:`Working

Reviewer: ```suggestion
- :guilabel:`Allocated Time`: Is calculated based on the date and the employee's :guilabel:`Working
```

  Schedule`.

> - :guilabel:`Role`: Select the role that the resources assigned will be performing. This field is

Reviewer: ```suggestion
- :guilabel:`Role`: Select the role that the assigned resources will be performing. This field is
```

>   used when :ref:`auto-planning <planning/open-shifts>` shifts. Once you select a role, the shift

Reviewer: But is not only used when auto-planning shifts, right? I would rephrase this. For example:

```suggestion
  especially useful when :ref:`auto-planning <planning/open-shifts>` shifts. Once you select a role, the shift
```

  templates associated with it are displayed in the upper section of the pop-up window.

Additionally, the following fields may also be available:

> - :guilabel:`Project`: If the :guilabel:`Project planning` setting is enabled, select the relevant

Reviewer: ```suggestion
- :guilabel:`Project`: If the :guilabel:`Project Planning` setting is enabled in the Planning app settings, select the relevant
```
Or similar to make clear where the setting can be enabled.

  project to link to the shift. Shifts dedicated to the selected project are then planned and
  tracked accordingly.
- :guilabel:`Task`: If a project is selected, optionally select a task within that project.
- :guilabel:`Sales Order`: If the Sales app is installed, select the relevant sales order to link to
  the shift.
- :guilabel:`Customer`: If the :doc:`Field Service <planning/field_service>` setting is enabled and

>   the shift is an onsite intervention, select the relevant customer to link to the shift.

Reviewer: ```suggestion
  the shift is an on-site intervention, select the relevant customer to link to the shift.
```

- :guilabel:`Worksheet`: If :ref:`field service worksheets <planning/field-service/worksheets>` are
  configured, optionally select the relevant worksheet template.

Click :guilabel:`Save` to create the shift, or :guilabel:`Publish` to create it and notify the
assigned employee by email.

.. _planning/shifts-sales-orders:

Create shifts from sales orders
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~

Shifts are created automatically from sales orders containing products configured as follows:

- :guilabel:`Product Type`: Set to :guilabel:`Service`.
- :guilabel:`Plan Services`: Enable the checkbox, then select the relevant :ref:`role
  <planning/roles>`.

Once the sales order is confirmed, schedule the shift:

> #. Click the :guilabel:`To plan` smart button on the sales order form.

Reviewer: ```suggestion
#. Click the :guilabel:`To Plan` smart button on the sales order form.
```

#. Click the relevant day in the appropriate resource's schedule.

   .. note::
      Only :ref:`resources whose role <planning/configuration>` matches the role configured for the
      product on the sales order are available for shift assignment.

#. Select the relevant shift from the list of open shifts for that role.
#. :ref:`Publish <planning/shifts-publish>` the shift.

.. tip::
   - To edit the shift before planning it, on the :guilabel:`Schedule by Resource` view:

     - either click :guilabel:`New`;
     - or click the relevant day in the appropriate resource's schedule, then click
       :guilabel:`Create New`.

>      Then, :ref:`edit the fields <planning/create-shift>` as needed, then :guilabel:`Save` the

Reviewer: ```suggestion
     Then, :ref:`edit the fields <planning/create-shift>` as needed, and :guilabel:`Save` the
```

- Extend a shift over several days by hovering over the right edge of the shift block. When the
  cursor changes into a resize pointer, click and drag the edge to the desired day.
- Split a shift spanning multiple consecutive days into two segments by hovering over the boundary
  between the two affected day columns, then clicking the :icon:`fa-scissors` (:guilabel:`scissors`)
  icon.

   .. image:: planning/split-shifts.png
      :alt: Split shifts tool.

To delete a shift, click it in the schedule, then click :guilabel:`Delete`.

.. tip::
   Alternatively, the shift form can be accessed by switching to the list view and clicking
   :guilabel:`View` on the relevant shift's line, or by switching to the Kanban view and clicking
   the relevant shift's card.

.. _planning/shifts-publish:

Publish shifts
--------------

Publishing shifts notifies the assigned employees by email and makes the schedule visible to
all users, including those without :guilabel:`Administrator` Planning :doc:`access rights
</applications/general/users/access_rights>` (via :menuselection:`Planning --> Planning -->
All Planning`).

> Shifts can be published individually or en masse.

Reviewer: ```suggestion
Shifts can be published individually or in bulk.
```

To publish an individual shift:

- either click :guilabel:`Publish` at the end of the :ref:`shift creation process
  <planning/create-shift>`;
- or click the shift in the schedule, click :guilabel:`Edit`, then click :guilabel:`Publish` on
  the shift form.

To publish all planned shifts:

#. From any schedule, click :guilabel:`Publish`.

> #. In the popup, optionally edit the :guilabel:`Period`.

Reviewer: ```suggestion
#. In the pop-up, optionally edit the :guilabel:`Period`.
```

#. Disable :guilabel:`Include Open Shifts` to exclude :ref:`open shifts <planning/open-shifts>` from
   the publication.
#. Optionally edit the list of employees; removing an employee from the list excludes their shifts
   from this publication.
#. Click :guilabel:`Publish & Send`.

Employees are notified of their shifts once they are published:

- Employees without user accounts are redirected to a dedicated **Planning portal**.
- Employees with a user account are redirected to the :guilabel:`My Planning` view.

.. note::
   - If the email address of one or more employees in the list has not been configured, enter their

>      :guilabel:`Work Email` in the popup that opens, then click :guilabel:`Save & Send Schedule`. If

Reviewer: ```suggestion
     :guilabel:`Work Email` in the pop-up that opens, then click :guilabel:`Save & Send Schedule`. If
```
